test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file FakeTimeStampDataGenerator.cpp FakeTimeStampDataGenerator class
+ * implementation
+ *
+ * This is part of the DUNE DAQ Software Suite, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "CommonIssues.hpp"
+#include "FakeTimeStampDataGenerator.hpp"
+
+#include "appfwk/app/Nljs.hpp"
+
+#include "logging/Logging.hpp"
+
+#include <chrono>
+#include <cstdlib>
+#include <thread>
+#include <string>
+#include <vector>
+
+using pd_clock = std::chrono::duration<double, std::ratio<1, 50000000>>;
```

Comment:
We'll have to handle the 50 MHz clock in PD-I and the 62.5 MHz clock in PD-II/DUNE, so the clock frequency will have to become a configuration parameter. I think that rules out using a `std::chrono::duration` like this (which needs to be defined at compile-time.

Reply:
I'm not familiar with chrono library. This way of implementation comes from the first version of the module. I will keep reading about it, but in the meantime, any suggestions on how to make it properly?

Reply:
You'll have to make the clock frequency a configuration parameter via the schema file, and use the frequency when you calculate the timestamp of your TSD. The `TriggerDecisionEmulator` in `trigemu` does something like this. The schema is:

https://github.com/DUNE-DAQ/trigemu/blob/develop/schema/trigemu/triggerdecisionemulator.jsonnet

and the plugin that uses it is 

https://github.com/DUNE-DAQ/trigemu/blob/develop/plugins/TriggerDecisionEmulator.cpp

Reply:
I made the frequency a configuration parameter, but still depending on chrono. I would need a bit more time to understand and use TriggerDecisionEmulator as an example.